torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -111,36 +126,126 @@ def create_subclass_meta(
+        tensors_and_sizes: List[Union[Tensor, SymInt]] = []
+
+        for attr in attrs:
+            inner_tensor = getattr(t, attr)
+            tensors_and_sizes.extend(flatten_subclass(inner_tensor))
+
+        if append_symints:
+            tensors_and_sizes.extend(filter_symints(t.size()))
+            tensors_and_sizes.extend(filter_symints(t.stride()))
+
+        return tensors_and_sizes
+
+    xs_inner: List[Union[int, Tensor, SymInt]] = []
+
+    for x in wrapped_args:
+        xs_inner.extend(flatten_subclass(typing.cast(Tensor, x)))
+
+    return xs_inner
+
+
+def runtime_unwrap_tensor_subclasses(
+    wrapped_args: List[Union[Tensor, int]],
+    *,
+    append_symints: bool,
+    subclass_metas: Optional[List[Union[int, SubclassCreationMeta]]] = None,
+):
+    def flatten_subclass(t: Union[Tensor, int], meta: Optional[SubclassCreationMeta]):
+        if not is_traceable_wrapper_subclass(t):
+            return [t]
+
+        assert isinstance(t, Tensor)
+
+        attrs, _ = t.__tensor_flatten__()
+        tensors_and_sizes: List[Union[Tensor, int]] = []
+
+        for attr in attrs:
+            inner_tensor = getattr(t, attr)
+            inner_meta = meta.attrs.get(attr)
+            tensors_and_sizes.extend(flatten_subclass(inner_tensor, inner_meta))
+
+        if append_symints:
+            assert isinstance(meta, SubclassCreationMeta)
+            # outer_size
+            size = t.size()
+            symint_placeholders = compute_symint_placeholders(meta.outer_size)
+            assert len(size) == len(symint_placeholders)
+            tensors_and_sizes.extend(
+                [r for (r, is_symint) in zip(size, symint_placeholders) if is_symint]
+            )
+
+            # outer_stride
+            stride = x.stride()
+            symint_placeholders = compute_symint_placeholders(meta.outer_stride)
+            assert len(stride) == len(symint_placeholders)
+            tensors_and_sizes.extend(
+                [r for (r, is_symint) in zip(stride, symint_placeholders) if is_symint]
+            )
+        return tensors_and_sizes
+
+    xs_inner: List[Union[int, Tensor, SymInt]] = []
+
+    for idx, x in enumerate(wrapped_args):
+        if not is_traceable_wrapper_subclass(x):
+            xs_inner.append(x)
+            continue
+
+        if subclass_metas is None:
+            xs_inner.extend(get_plain_tensors(typing.cast(Tensor, x)))
+        else:
+            meta = subclass_metas[idx]
+            assert isinstance(meta, SubclassCreationMeta)
+            xs_inner.extend(flatten_subclass(typing.cast(Tensor, x), meta))
+
+    return xs_inner
 
 
 def unwrap_tensor_subclasses_with_indices_to_original(wrapped_args):
     ret_unwrapped = []
     ret_indices_to_original = []
     for i, a in enumerate(wrapped_args):
-        a_unwrapped = unwrap_tensor_subclasses([a], is_joint_structure=False)
+        a_unwrapped = runtime_unwrap_tensor_subclasses([a], append_symints=False)
```

Comment:
why are we calling the `runtime_*` variant here? grepping around, it looks like `unwrap_tensor_subclasses_with_indices_to_original` is only called at trace time